### PR TITLE
feat: Implement `getTokenURI` from IERC721

### LINKED ISSUE #130: [Core] Implement HTS emulation for NFT methods
In addition to ERC20, HTS contract should provide emulation for NFT/ERC721 methods. These methods are specified in [HIP-218](https://hips.hedera.com/hip/hip-218).

### PR DESCRIPTION
**Pull request by victor-yanev**
**Description**:

This PR implements the core functionality of `getTokenURI` selector from `ERC-721`

**Related issue(s)**:

Fixes #130 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: contracts/HtsSystemContract.sol [hunk 1]
```diff
@@ -395,6 +401,11 @@ contract HtsSystemContract is IHederaTokenService, IERC20Events, IERC721Events {
         assembly { amount := sload(slot) }
     }
 
+    function __tokenURI(uint256 serialId) private returns (string memory uri) {
+        bytes32 slot = _tokenUriSlot(uint32(serialId));
+        uri = readString(slot);
```
Comment (acuarica) [inline contracts/HtsSystemContract.sol:406]: Let's see if this works, it points `_uri` to the dynamic slot so we can read using usual Solidity.

```suggestion
        string storage _uri;
        assembly { _uri.slot := slot }
        uri = _uri;
```

**Review by acuarica — COMMENTED** [review]
Nice work, left a comment. If we managed to change the implementation, we avoid re-implementing Solidity strings.

File: contracts/HtsSystemContract.sol (hunk 1 above)
Comment (victor-yanev) [reply contracts/HtsSystemContract.sol]: Great suggestion! Yes, it works!

**Review by acuarica — APPROVED** [review]
Nice, great stuff!

**Pull request merged**